### PR TITLE
untwist bad logic, tidy code, revive save of local geometry

### LINKED ISSUE #308: Writting of local geometry is broken
## Description
The writing of the local geometry in the "grigen.x" application does not work anymore. 

### To Reproduce:
1. Trigger the writing of the local geometry when running gridgen.x: Add this line to the Geometry section:  `save_local_domain: 1`
2. mpirun -np manypes soca_grigen.x gridgen.yml
3. Run fails in a netcdf abort trap ...

### LINKED ISSUE #346: Clean-up of unused variable in geometry
## Description
Some twisted logic in Geometry and "read_soca_grid" is unused and cluttering most of the yaml files.  

## Solution
Untwist and remove unused variable!

### PR DESCRIPTION
**Pull request by guillaumevernieres**
## Description
Cleanup of poor logic and removal of unused variables in the geometry setup and yaml files. While a lot of files have been touched, code changes are confined to `soca_geom_mod.F90`. 

**Provide a detailed description of what this PR does.**  
- Removed [Unused logic/variable](https://github.com/JCSDA/soca/blob/4462283365ed9904a7fbe0d698abd8519a72b678/src/soca/Geometry/soca_geom_mod.F90#L109)
- Revived writing of local geometry, useful for debugging and making nostalgic disco plots:

fms-domains

**What bug does it fix, or what feature does it add?**
local geometry writing was broken, it is now fixed.

**Is a change of answers expected from this PR?**
**NO**

### Issue(s) addressed
- closes #346 
- closes #308 

## Testing

How were these changes tested? **ctest only on(in?) containers**
What compilers / HPCs was it tested with? **gcc7.4**
Are the changes covered by ctests? **Yes, local writing of geometry has been added to the gridgen test**

## Dependencies
**NONE**

### CODE REVIEW

**Comment by guillaumevernieres** [comment]
An other great success story! @travissluka  @clouden90 , don't waste your time yet, after rebuilding/cleaning, that pr doesn't work anymore.

**Comment by guillaumevernieres** [comment]
@travissluka , @clouden90 : Ready for review ... Sorry about the minor snafu. Did you see that coverage!

**Review by travissluka — APPROVED** [review]
great, i'll have to add disco mode to the ioda-plots

File: test/testinput/gridgen.yml [hunk 1]
```diff
@@ -3,7 +3,8 @@ geometry:
   num_ice_lev: 4
   num_sno_lev: 1
   geom_grid_file: soca_gridspec.nc
-
+  save_local_domain: 1
+  full_init: 1
```
Comment (aerorahul) [inline test/testinput/gridgen.yml]: This is declared as a boolean in the code. Make it logical here too.

**Review by clouden90 — APPROVED** [review]
Fantastic! Amazing code coverage.

File: src/soca/Geometry/soca_geom_mod.F90 [hunk 2]
```diff
@@ -106,8 +107,12 @@ subroutine geom_init(self, f_conf, f_comm)
   ! Allocate geometry arrays
   call geom_allocate(self)
 
-  if ( f_conf%has("read_soca_grid") ) &
-      call geom_read(self)
+  ! Check if a full initialization is requiered, default to false
```
Comment (aerorahul) [inline src/soca/Geometry/soca_geom_mod.F90]: required.

File: src/soca/Geometry/soca_geom_mod.F90 [hunk 3]
```diff
@@ -106,8 +107,12 @@ subroutine geom_init(self, f_conf, f_comm)
...
+  if ( .not. f_conf%get("full_init", full_init) ) full_init = .false.
+
+  ! Read the geometry from file by default,
+  ! skip this step if a full init is requiered
```
Comment (aerorahul) [inline src/soca/Geometry/soca_geom_mod.F90]: not requiered

**Review by aerorahul — CHANGES_REQUESTED** [review]
mis-spelled comments.
inconsistent datatype declaration and use.

**Comment by guillaumevernieres** [comment]
Thanks for the review @aerorahul will work on the changes later today.

**Review by aerorahul — APPROVED** [review]
that wasn't too hard, was it?

**Comment by guillaumevernieres** [comment]
> that wasn't too hard, was it?

I haven't quite yet recovered. Thanks for your review @aerorahul. :)

**Pull request merged**